buildSrc/subprojects/build/src/main/groovy/org/gradle/build/docs/AssertNoDaemonAsciidoctorTask.groovy

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradle.build.docs
+
+import org.gradle.internal.environment.GradleBuildEnvironment
+import javax.inject.Inject
+import org.asciidoctor.gradle.AsciidoctorTask
+
+class AssertNoDaemonAsciidoctorTask extends AsciidoctorTask {
```

Comment:
Can we add a custom action to the `AsciidoctorTask` instead of subclassing it? Something like
```
tasks.withType(AsciidoctorTask) {
    doFirst {
        assert !buildEnvironment.longLivingProcess
    }
}
```

Reply:
The only issue is we have to fetch `buildEnvironment` for the build. Thanks I'll try it.